### PR TITLE
Scene Refactor: Make inputs exclusive

### PR DESCRIPTION
**Pull request by Endaris**
Implements #1099 .
The top level input table no longer includes inputs from input configurations that have been claimed by players *besides* Menu identifiers like `MenuSelect` etc. as these are meant to be player unspecific.
The input table now offers specific functions to claim and release input configurations and will take immediately update the top level input table to reflect the changes.

I tried to adapt variable names a bit so it's easier to read and made it so that the standard is having *no* inputConfiguration explicitly when none was claimed as that always felt a bit confusing.

### CODE REVIEW

**Comment by Endaris** [comment]
I think not being able to change controllers between puzzles is fine.
It should be possible between 1p vs matches though and it was possible before his PR so I'll address this here instead of an issue.

**Comment by Endaris** [comment]
I managed to fix it for regular 1p modes but I introduced crashes for puzzle mode with that.
The PuzzleGame scene currently entirely bypasses the BattleRoom update loop and creates and starts matches on its own terms in order to go from puzzle to puzzle directly and maintain the stage + music without restarting it.
This means that no hooks are set for BattleRoom functionality so that when starting the second puzzle set in succession, it tries to lock a second input configuration for the player because the old one never got released.
I can probably wing this by imitating some part of BattleRoom functionality within PuzzleGame but ideally it should get changed to accomodate the standard BattleRoom loop instead.
tl;dr this is tricky and will take a bit

**Comment by Endaris** [comment]
Essentially my idea is to move music control from Match to GameBase and have Match emit signals for danger music changes.
That way the puzzle game scene can consume signals from match about match end and danger music change. That way the continuous music playback for puzzle mode can be realized without circumventing the BattleRoom loop.
I already have a branch that moves the music updates over in this manner but I'll have to look at this again and see how PuzzleGame can work with BattleRoom nicely now that this main problem is out of the way.

**Comment by Endaris** [comment]
I think this is overall okay now.
I'll try to work on the toolchain for canary releases next (even if they might run over releases on my fork for starters to test things out) so bigger changes like this can get - while not necessarily code reviewed - at least get tested by players without affecting the beta build.
